std/assembly/console.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  process
+} from "./process";
+
+// @ts-ignore: decorator
+@lazy var timers = new Map<string,u64>();
+
+export namespace console {
+
+  export function assert<T>(condition: T, message: string): void {
+    if (!condition) {
+      let stderr = process.stderr;
+      stderr.write("Assertion failed: ");
+      stderr.write(message);
+      stderr.write("\n");
+    }
+  }
+
+  export function log(message: string): void {
+    var stdout = process.stdout;
+    stdout.write(message);
+    stdout.write("\n");
+  }
+
+  export function debug(message: string): void {
+    var stdout = process.stdout;
+    stdout.write("Debug: ");
+    stdout.write(message);
+    stdout.write("\n");
+  }
+
+  export function info(message: string): void {
+    var stdout = process.stdout;
+    stdout.write("Info: ");
+    stdout.write(message);
+    stdout.write("\n");
+  }
+
+  export function warn(message: string): void {
+    var stdout = process.stdout;
+    stdout.write("Warning: ");
+    stdout.write(message);
+    stdout.write("\n");
+  }
+
+  export function error(message: string): void {
+    var stdout = process.stdout;
+    stdout.write("Error: ");
+    stdout.write(message);
+    stdout.write("\n");
+  }
+
+  export function time(label: string): void {
+    var stdout = process.stdout;
+    if (timers.has(label)) {
+      stdout.write("Duplicate label '");
+      stdout.write(label);
+      stdout.write("' for console.time()");
+      stdout.write("\n");
+      return;
+    }
+    timers.set(label, process.hrtime());
+  }
+
+  export function timeLog(label: string): void {
+    var stdout = process.stdout;
+    if (!timers.has(label)) {
+      stdout.write("No such label '");
+      stdout.write(label);
+      stdout.write("' for console.timeLog()");
+      stdout.write("\n");
+      return;
+    }
+    timeLogImpl(label);
+  }
+
+  export function timeEnd(label: string): void {
+    var stdout = process.stdout;
+    if (!timers.has(label)) {
+      stdout.write("No such label '");
+      stdout.write(label);
+      stdout.write("' for console.timeEnd()");
+      stdout.write("\n");
+      return;
+    }
+    timeLogImpl(label);
+    timers.delete(label);
+  }
+}
+
+function timeLogImpl(label: string): void {
+  var start = changetype<u64>(timers.get(label));
+  var end = process.hrtime();
+  var nanos = end - start;
+  var millis = nanos / 1000000;
+  var millisStr = millis.toString();
+  var stdout = process.stdout;
+  stdout.write(label);
+  stdout.write(": ");
+  stdout.write(millisStr);
+  stdout.write("ms");
+  stdout.write("\n");
```

Comment:
```suggestion
  stdout.write("ms\n");
```

Reply:
Yeah, still wondering how much it's worth to make use of `writeStringFast`. In this particular call, due to the strings being two static latin1 chars each, we get two times `writeStringFast` instead of a redundant encode with garbage. May be worth it here, but not so much in the other cases.

Reply:
ok